src/NvimClient.APIGenerator/NvimAPIGenerator.cs

```diff
@@ -123,11 +123,28 @@ private object GetExtensionType(MessagePackExtendedTypeObject msgPackExtObj)
       string.Join("\n",
         uiEvents.Select(uiEvent =>
         {
-          var camelCaseName = StringUtil.ConvertToCamelCase(uiEvent.Name, true);
+          var camelCaseName = $"{StringUtil.ConvertToCamelCase(uiEvent.Name, true)}Event";
           var genericTypeParam = uiEvent.Parameters.Any()
-            ? $"<{camelCaseName}EventArgs>"
+            ? $"<{camelCaseName}Args>"
             : string.Empty;
-          return $"    public event EventHandler{genericTypeParam} {camelCaseName}Event;";
+          return $@"
+    /// <summary>
+    /// EventHandler for <c>{uiEvent.Name}</c> UI event (see corresponding
+    /// docs in `:help ui-events` in nvim).
+    /// </summary>
+    /// <example>
+    /// <code>
+    /// var api = new NvimAPI();
+    /// api.{camelCaseName} += (sender, args) =>
+    /// {{
+    ///     // `args` {(genericTypeParam == string.Empty ? "contains no data for this event." : $"is of type `{camelCaseName}Args`.")}
+    ///     // Handler code goes here.
+    /// }}
+    /// // Now if this event is emitted after attaching the UI,
+    /// // the above handler code will be run.
```

Comment:
maybe move the comment above the handler definition, and...

```suggestion
    /// // This handler will be executed whenever the event is emitted after attaching the UI.
```

Reply:
👍 just unresolved this until I move the comment, then I’ll merge.